File: front/src/app/classes/pagination.ts
import { StatusTransaction } from '../enums/status-transaction.enum';
import { TypeTransaction } from '../enums/type-transaction.enum';

export class Pagination<T> {
  public data: Array<T>;

  public pageNumber: number;

  public pageSize: number;

  public totalCount: number;

  public pageCount: number;

  public sortStatusBy: StatusTransaction;

  public sortTypeBy: TypeTransaction;
}

File: backend/WebAPI/Classes/Pagination.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Entities.Enums;
using WebAPI.Attributes.Validators;

namespace WebAPI.Classes
{
  public class Pagination<T>
  {
    public IEnumerable<T> Data { get; set; }

    [Required, Range(0, int.MaxValue)]
    public int PageNumber { get; set; }

    [Required, Range(1, 100)]
    public int PageSize { get; set; }

    public int? TotalCount { get; set; }

    public int? PageCount { get; set; }

    public EnumStatusTransaction? SortStatusBy { get; set; }

    public EnumTypeTransaction? SortTypeBy { get; set; }
  }
}